Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -2424,7 +2425,7 @@ template <class TreeBuilder> bool Parser<LexerType>::parseFunctionInfo(TreeBuild
 
             functionInfo.body = context.createFunctionMetadata(
                 startLocation, endLocation, startColumn, bodyEndColumn, 
-                functionKeywordStart, functionNameStart, parametersStart, 
+                functionKeywordStart, functionNameStart, parametersStart, implementationVisibility(),
```

Comment:
@Constellation should I add this to `SourceProviderCacheItem`?  Or is it fine for this to use the current `Scope`?

Reply:
I think this is fine since this status can be computed from the current scope. But it is nice to leave a comment about it here.